### ISSUE TITLE
swxftlint:disable:next fights with swift comments

### ISSUE THREAD

**Issue by deanWombourne** [realm/SwiftLint#2409]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

If I want to both comment a type, and disable the type_name warning, I can only choose one or the other. e.g.

```swift
// swiftlint:disable:next type_name
/// Some swift comment for the type `WhatAVeryLongTypeNameThisWillNotLint`
enum WhatAVeryLongTypeNameThisWillNotLint { }
```
This will return the `Superfluous Disable Command Violation`, and the warning for the long type name.

```swift
/// Some swift comment for the type `WhatAVeryLongTypeNameThisWillNotLint`
// swiftlint:disable:next type_name
enum WhatAVeryLongTypeNameThisWillNotLint { }
```
This will compile, but the comments now aren't associated with the type.

I can't think of a good way round this, currently I'm just enabling and disabling the rule around that type so it's not a massive issue, but you might have a better idea :)

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ Pods/SwiftLint/swiftlint --path ..
Linting '...' (../249)
Classes/Model/WhatAVeryLongTypeNameThisWillNotLint.swift:41:6: warning: Type Name Violation: Type name should be between 3 and 40 characters long: 'WhatAVeryLongTypeNameThisWillNotLint' (type_name)
Classes/Model/WhatAVeryLongTypeNameThisWillNotLint.swift:40: warning: Superfluous Disable Command Violation: SwiftLint rule 'type_name' did not trigger a violation in the disabled region. Please remove the disable command. (superfluous_disable_command)
Done linting! Found 2 violations, 0 serious in 249 files.
```

### Environment

* SwiftLint version: 0.27.0
* Installation method used:  CocoaPods
* Paste your configuration file: Reproducible without config file
* Which Xcode version are you using: 9.4.1
* Do you have a sample that shows the issue? See examples in description

**Comment by marcelofabri** [comment]
This is expected because `disable:next` only disables violations on the next line. You can use `disable:this` on the same line or `disable:previous` on the line after the declaration.

Let me know if that works for you!
